src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -149,53 +143,32 @@ void RandomFourierGaussPreproc::set_dim_feature_space(const int32_t dim) {
 				"void CRandomFourierGaussPreproc::set_dim_feature_space(const int32 dim): dim<=0 is not allowed");
 	}
 
+	is_updated = dim_feature_space == dim;
```

Comment:
should be `is_updated = False`. If a user changes the dimensions, it should result in having to re-compute stuff

Reply:
What if the user ended up putting the same dim as before?, then we would've wrongly put is_updated = False. 
Instead, we check if it's same as the previous, if not then make is_updated to false